### ISSUE TITLE
[Mobile Payments] There is no OAuth Connection between your platform and (account) in test mode

### ISSUE THREAD

**Issue by allendav** [Automattic/woocommerce-payments#1674]
I have a live site. If I enable test mode on it, I get the following error when the Woo mobile app (iOS or Android) attempts to get a card reader connection token through the plugin's `'payments/connection_tokens` endpoint:

```
Error: There is no OAuth Connection between your platform and acct_(REDACTED) in test mode. See https://stripe.com/docs/connect/standard-accounts for details on how to establish a connection.
```

If I uncheck Enable test mode, token fetch works successfully.

It also appears that in at least one case, this varies based on which WordPress.com user is logged into the Woo mobile app.

cc @malinajirka @ctarda

**Comment by allendav** [comment]
Some more insights. I compared a WooCommerce Payments account of mine (ending in ...8tHB) which was created using our platform's live key with another WooCommerce Payments account of mine for another site (ending in ...3m18) which was created using our platform's testmode key.

- Attempting to fetch a reader connection token for ...8tHB with our platform's live key succeeds.
- Attempting to fetch a reader connection token for ...8tHB with our platform's testmode key fails with the error noted above.
- Attempting to fetch a reader connection token for ...3m18 with our platform's testmode key succeeds.

I've reached out to Stripe for insights.
